app/controllers/host/vms_controller.rb

```diff
@@ -0,0 +1,28 @@
+class Host::VmsController < ApplicationController
```

Comment:
I think you & @ohadlevy  may have missed `ComputeResourcesVmsController`. This controller already implements these `find` methods, and could have a `import` method that implements the `new` method I see in this controller.